### PR TITLE
[compat] Remove compat with legacy python/django versions

### PR DESCRIPTION
**Pull request by Djailla**

### CODE REVIEW

File: whitenoise/storage.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from __future__ import absolute_import, print_function
+from __future__ import absolute_import
```
Comment (evansd) [inline whitenoise/storage.py]: Both future imports should be remove here

File: whitenoise/string_utils.py [hunk 2]
```diff
@@ -1,12 +1,7 @@
...
 
-
-if sys.version_info[0] >= 3:
-    BINARY_TYPE = bytes
-    TEXT_TYPE = str
-else:
-    BINARY_TYPE = str
-    TEXT_TYPE = unicode  # noqa: F821
+BINARY_TYPE = bytes
+TEXT_TYPE = str
```
Comment (evansd) [inline whitenoise/string_utils.py]: These constants are redundant now and references to them should be just be replaced with `bytes` and `str`

File: whitenoise/string_utils.py [hunk 3]
```diff
@@ -17,19 +12,8 @@ def decode_if_byte_string(s, force_text=False):
     return s
 
 
-# Follow Django in treating URLs as UTF-8 encoded (which requires undoing the
-# implicit ISO-8859-1 decoding applied in Python 3). Strictly speaking, URLs
-# should only be ASCII anyway, but UTF-8 can be found in the wild.
```
Comment (evansd) [inline whitenoise/string_utils.py]: This comment is still required and should be retained

File: whitenoise/string_utils.py (hunk 2 above)
Comment (Djailla) [reply whitenoise/string_utils.py]: Done

**Comment by Djailla** [comment]
Hello @evansd any update for this PR ? Thanks 👍

**Comment by evansd** [comment]
:+1: Thanks so much for this!

**Pull request merged**